avocado/utils/distro.py

```diff
@@ -331,6 +331,50 @@ class FedoraProbe(RedHatProbe):
     CHECK_VERSION_REGEX = re.compile(r'Fedora release (\d{1,2}).*')
 
 
+class AmazonLinuxProbe(Probe):
+
+    """
+    Probe for Amazon Linux systems
+    """
+
+    CHECK_FILE = '/etc/os-release'
+    CHECK_FILE_CONTAINS = 'Amazon Linux'
+    CHECK_FILE_DISTRO_NAME = 'amzn'
+
+    def get_distro(self):
+        dst = Probe.get_distro(self)
+
+        if not dst.name == self.CHECK_FILE_DISTRO_NAME:
+            return dst
+
+        version_re = re.compile(r'VERSION="(.*)"')
+        version = None
+
+        with open(self.CHECK_FILE) as check_file:
+            for line in check_file:
+                match = version_re.match(line)
+                if match:
+                    version = match.group(1)
+
+        if version:
+            version_parts = version.split()
+            if len(version_parts) == 1:
+                dst.version = 1
+                try:
+                    dst.release = re.findall(r'\d*\.\d*', version_parts[0])[0]
+                except IndexError:
+                    dst.release = UNKNOWN_DISTRO_RELEASE
+            else:
+                try:
+                    dst.version = int(version_parts[0].split('.')[0])
+                    dst.release = re.findall(r'\d*\.\d*', version_parts[1])[0]
```

Comment:
Here you only look at the first 2 items. How about using `split(None, 1)` to avoid leaving some extra version items behind?

Anyway if it was me I'd simply use `split(None, 1)` and either set `dst.version` or `dst.version|release` based on number of items returned. Is it necessary to enforce `\d*,\d*` format? Don't take me wrong, I like it's specific, but in the future it might require additional tweaks...

PS: No need to change anything, simple comment that you intend to ignore the remaining version items behind is fine. These are just recommendations to avoid false-positives.

Reply:
It's mostly because the version numbers are like 2017.09 or similar. I'll consider your suggestions.